src/runner/main.cpp

```diff
@@ -291,6 +292,50 @@ toast_notification_handler_result toast_notification_handler(const std::wstring_
     }
 }
 
+void cleanup_updates()
+{
+    auto state = UpdateState::read();
+    if (state.state != UpdateState::upToDate)
+    {
+        return;
+    }
+
+    // Msi and exe files
+    for (const auto& entry : std::filesystem::directory_iterator(updating::get_pending_updates_path()))
+    {
+        auto entryPath = entry.path().wstring();
+        std::transform(entryPath.begin(), entryPath.end(), entryPath.begin(), ::towlower);
+
+        if (entryPath.ends_with(L".msi") || entryPath.ends_with(L".exe"))
+        {
+            std::error_code err;
+            std::filesystem::remove(entry, err);
+            if (err.value())
+            {
+                Logger::warn("Failed to delete installer file {}. {}", entry.path().string(), err.message());
+            }
+        }
+    }
+	
+    // Log files
+    auto rootPath{ PTSettingsHelper::get_root_save_folder_location() };
+    auto currentVersion = left_trim<wchar_t>(get_product_version(), L"v");
+    for (const auto& entry : std::filesystem::directory_iterator(rootPath))
+    {
+        auto entryPath = entry.path().wstring();
+        std::transform(entryPath.begin(), entryPath.end(), entryPath.begin(), ::towlower);
+        if (entry.is_regular_file() && entryPath.ends_with(L".log") && entryPath.find(currentVersion) == std::string::npos)
+        {
+            std::error_code err;
+            std::filesystem::remove(entry, err);

```

Comment:
Does std::filesystem::directory_iterator work well with us deleting entries while it's iterating?

Reply:
It should be fine since similar code was already used in PowerToys.Update to cleanup msi and exe files and I haven't experienced any issues during tests.
From docs: https://en.cppreference.com/w/cpp/filesystem/directory_iterator
- The iteration order is unspecified, except that each directory entry is visited only once.
- If a file or a directory is deleted or added to the directory tree after the directory iterator has been created, it is unspecified whether the change would be observed through the iterator.